api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java

```diff
@@ -31,6 +31,7 @@
 	private Integer minLength; // If > 0, will always return identifiers with a minimum of this length
 	private Integer maxLength; // If > 0, will always return identifiers no longer than this length
     private String baseCharacterSet; // Enables configuration in appropriate Base
+	private Integer nextSequenceValue; // Pre-generate the next identifier
 
```

Comment:
Yes, we specifically did not include the nextSequenceValue here because we wanted to avoid using Hibernate to access this property because of synchronization issues--we need to guarantee that the same number isn't issued twice.  I'm pretty sure we do not want to add this, but rather should be using the getSequenceValue and saveSequenceValue service methods to get and set the sequence value.  This may require a custom REST call, I suppose.


How is it working in your use case?